libs/plugin/src/executors/sync-repo/executor.ts

```diff
@@ -0,0 +1,43 @@
+import type { SyncRepoExecutorSchema } from './schema'
+import { workspaceRoot } from '@nx/devkit'
+import { execSync } from 'child_process'
+import * as fs from 'fs'
+import * as path from 'path'
+
+export default async function runExecutor(
+  options: SyncRepoExecutorSchema
+): Promise<{ success: boolean }> {
+  const { localPath, repoUrl: url, prefixPath: prefix = true } = options
+  const cwd = prefix ? path.join(workspaceRoot, localPath) : localPath
+  const name = path.basename(cwd)
+
+  if (fs.existsSync(cwd)) {
+    // Fetch & reset
+    const remoteHash = getRemoteRepoHash(url)
+    const localHash = getLocalRepoHash(cwd)
+    if (remoteHash !== localHash) {
+      console.log('Local and remote hashes differ, fetching')
+      execSync(`git fetch -q --depth 1`, { cwd })
+      execSync(`git reset -q --hard origin/master`, { cwd })
```

Comment:
Do we need these to be quiet? it would make it harder to debug on failure and it should rarely run anyway so it shouldnt spam the console
```suggestion
      execSync(`git fetch --depth 1`, { cwd })
      execSync(`git reset --hard origin/master`, { cwd })
```

Reply:
The output texts are gonna be cached. So I'm not sure if we should include volatile texts from git. If you insist, I can un-quiet them.